utils/username.ts

```diff
@@ -0,0 +1,29 @@
+const { MAX_USERNAME_LENGTH } = require("../constants/users");
+
+export const formatUsername = (firstName: string, lastName: string, suffix: number): string => {
+    const trimmedFirstName = firstName.trim();
+    const trimmedLastName = lastName.trim();
+
+    const firstNamePart = trimmedFirstName.split(/[\s-]/)[0];
+    const validFirstName = /^[a-zA-Z]+$/.test(firstNamePart)
+        ? firstNamePart.toLowerCase()
+        : "null";
+
+    const lastNameParts = trimmedLastName.split(/[\s-]/);
+    const lastNamePart = lastNameParts[lastNameParts.length - 1];
+    const validLastName = /^[a-zA-Z]+$/.test(lastNamePart)
+        ? lastNamePart.toLowerCase()
+        : "null";
+
```

Comment:
What are the regex for?

Reply:
To validate the characters
for more, you can look format username tests